java/fury-core/src/main/java/org/apache/fury/reflect/TypeToken.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (C) 2006 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.fury.reflect;
+
+import com.google.common.reflect.TypeParameter;
```

Comment:
Could we implement `TypeToken` by ourself without depending on any guava API. `org.apache.fury.type.GenericType#build(java.lang.reflect.Type, java.util.function.Predicate<java.lang.reflect.Type>)` can be taken as an example.

Reply:
Fury use `TypeToken` to capture generic info and for Generic Type resolve only. We don't need all features of `TypeToken`

Reply:
For Generic Type resolve, Fury use `TypeToken` to:
- Resolve covariantType/invariantType
- Create new generic Type based on other generic Types:
```java
new TypeToken<Map<K, V>>() {}.where(new TypeParameter<K>() {}, keyType)
        .where(new TypeParameter<V>() {}, valueType);
```

Reply:
I need more time to go over the code and understand the usage properly. Can this be done in follow-up PR?